packages/compat/src/resolver-transform.ts

```diff
@@ -350,3 +358,51 @@ function handleComponentHelper(
 
   // resolver.unresolvableComponentArgument(componentName, argumentName, moduleName, param.loc);
 }
+
+function handleDynamicHelper(
+  param: ASTv1.Node,
+  resolver: Resolver,
+  moduleName: string
+  // scopeStack: ScopeStack,
+): void {
+  // let locator: ComponentLocator;
+  // switch (param.type) {
+  //   case 'StringLiteral':
+  //     locator = { type: 'literal', path: param.value };
+  //     break;
+  //   case 'PathExpression':
+  //     locator = { type: 'path', path: param.original };
+  //     break;
+  //   case 'MustacheStatement':
+  //     if (param.hash.pairs.length === 0 && param.params.length === 0) {
```

Comment:
it will be great to resolve super simple dynamic cases, like

```hbs
{{helper (if this.a "foo" "bar")}}
```

```hbs
{{helper (if this.localHelper this.localHelper "fallback-helper")}}
```

Reply:
@lifeart I think it was a deliberate choice not to support that for these helpers. https://github.com/embroider-build/embroider/issues/1018#issuecomment-999716009

Maybe I just misunderstood, but the RFC doesn't mention that either I think.

Reply:
Yes, the dynamic case can be dynamic *outside* of the `(helper )` call:

```hbs
{{if this.localHelper this.localHelper (helper "fallback-helper") }}
```

But also, since helpers are first-class values that can be passed around IMO it's nicer to implement your fallback-helper in the corresponding JS file anyway, so that it's not polluting the global namespace:

```hbs
{{if this.localHelper this.localHelper this.fallbackHelper }}
```

```js
import { helper } from '@ember/component/helper';
import Component from '@glimmer/component';
export default class extends Component {
  fallbackHelper = helper(function() {...});
}
```

Reply:
https://github.com/ember-template-lint/ember-template-lint/issues/2375

Reply:
I confirmed that ember already hard-enforces the main thing we care about here:

>  Assertion Failed: Passing a dynamic string to the `(helper)` keyword is disallowed. (You specified `(helper s)` and `s` evaluated into "my-helper".) This ensures we can statically analyze the template and determine which helpers are used. If the helper name is always the same, use a string literal instead, i.e. `(helper "my-helper")`. Otherwise, import the helpers into JavaScript and pass them to the helper keyword. See https://github.com/emberjs/rfcs/blob/master/text/0496-handlebars-strict-mode.md#4-no-dynamic-resolution for details. ('sample2/templates/application.hbs' @ L3:C9) 

The last bit of that message is actually out of date for Ember 3.25+ because you don't need to even pass them to the helper keyword, you can just use them as-is.

Reply:
Fired off a PR to update those out-of-date instructions in Ember https://github.com/emberjs/ember.js/pull/19950